geode-docs/basic_config/data_entries_custom_classes/managing_data_entries.html.md.erb

```diff
@@ -23,7 +23,24 @@ Program your applications to create, modify, and manage your cached data entries
 
 <a id="managing_data_entries__section_AACC36127F17411F86D1E409B86C6E5C"></a>
 **Note:**
-If you do not have the cache's `copy-on-read` attribute set to true, do not change the objects returned from the Java entry access methods. Instead, create a copy of the object, then modify the copy and pass it to the Java `put` method. Modifying a value in place bypasses the entire distribution framework provided by <%=vars.product_name%>, including cache listeners and expiration activities, and can produce undesired results.
+If you do not have the cache's `copy-on-read` attribute set to true, do not change the objects returned from the Java entry access methods. See [Safe Entry Modification](#managing_data_entries__section_A0E0F889AC344EFA8DF304FD64418809).
+
+## <a id="managing_data_entry_keys" class="no-quick-link"></a>Keys
+
+The `hashCode()` of the key for a data entry maps the entry
+within the region; no duplicates are permitted.
```

Comment:
This comment about hashcodes is a little confusing - it sounds like no duplicate hashcodes are permitted, which is not true. What are we trying to tell the user about their hashCode method here?

Reply:
Also, since we are stressing that hashCode must return the same value across servers, maybe we should also mention that equals should return the same result on different servers as well?